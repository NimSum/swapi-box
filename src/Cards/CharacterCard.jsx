import React from 'react';

export const CharacterCard = cards => 
  cards.map(character => (
    <article className="character-card">
      <h3>{ character.name }</h3>
      <h4>{ character.species }</h4>
      <div>
        <h4>{ character.homeworld }</h4>
        <p>{ character.population }</p>
      </div>
    </article>
  ))
