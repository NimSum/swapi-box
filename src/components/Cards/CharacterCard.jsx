import React from 'react';
import PropTypes from 'prop-types';
import { FavoriteCardBtn } from '../Buttons/Buttons';

export const CharacterCard = ({ card, updateFavoriteCount }) => 
  (
    <article key={ card.id } className="character-card">
      < FavoriteCardBtn 
       card={ card }
       updateFavoriteCount={updateFavoriteCount} />
      <h3>{ card.name }</h3>
      <h4>{ card.speciesName }</h4>
      <div className="home-planet">
        <h4>{ card.worldName }</h4>
        <p>{ card.worldPopulation }</p>
      </div>
    </article>
  )

CharacterCard.propTypes = {
  cards: PropTypes.array
}