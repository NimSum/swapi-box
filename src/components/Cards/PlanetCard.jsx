import React from 'react';

export const PlanetCard = cards => 
  <div>HIIII</div>
  // cards.map(planet => (
  //   <article className="planet-card">
  //     <h3>{ planet.name }</h3>
  //     <div className="residents">
        
  //     </div>
  //   </article>
  // ))