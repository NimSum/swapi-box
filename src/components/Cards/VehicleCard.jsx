import React from 'react';
import { FavoriteCardBtn } from '../Buttons/Buttons';

export const VehicleCard = ({ cards, updateFavoriteCount }) => 
  cards.map(vehicle => (
    <article className="vehicle-card">
      < FavoriteCardBtn 
        card={ vehicle }
        updateFavoriteCount={updateFavoriteCount}
        type="vehicle" />
      <h3>{ vehicle.name }</h3>
      <div className="vehicle-specs">
        <p>{ vehicle.model }</p>
        <p>{ vehicle.vehicle_class }</p>
        <p>{ parseInt(vehicle.passengers) + parseInt(vehicle.crew) }</p>
      </div>
    </article>
  ))
