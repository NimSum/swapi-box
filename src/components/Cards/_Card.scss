section.card-container {
  article {
    border-radius: 20px;
    width: 100%;
    min-width: 300px;
    margin: 0 auto;
    box-shadow: $card-shadow;
    position: relative;
    background-color: $card-bg-opacity;

    h3, h4, div, button, p {
      position: relative;
      z-index: 1;
    }

    h4 {
      font-family: $title-font;
      margin: 10px;
    }
    
    h4 {
      color: #f08a5d;
      font-size: 1.3rem;
    }

    h3 {
      color: $font-card-title-color;
      border-width: 2px 0 2px 0;
      border-style: solid;
      border-color: $card-title-border-color;
      padding: 10px 0;
      font-size: 1.7rem;
    }

    p {
      font-family: $content-font;
      margin: 5px auto;
      padding: 2px;
      width: 50%;
      color: $font-dark;
      background-color: $font-card-desc-color;
      border-radius: 10px;
    }

    img.bg-img {
      position: absolute;
      bottom: 0;
      z-index: 0;
      opacity: .1;
    }
  }


  article.character-card {
    max-height: 350px;

    div.home-planet {
      border-top: 2px solid $card-title-border-color;
      width: 50%;
      white-space: nowrap;
      margin: auto;

      p {
        width: 100%;
      }
    }

    img.bg-img {
      width: 200px;
      left: 0;
      opacity: .2;
    }
  }

  article.vehicle-card {
    min-height: 300px;
    max-height: 350px;
    img.bg-img {
      width: 100%;
      left: 0;
      opacity: .2;
    }
  }

  article.planet-card {
    min-height: 450px;
    max-height: 600px;
    overflow: scroll;

    img.bg-img {
      width: 90%;
      top: 15%;
      left: 0;
      margin: 0 5%;
      opacity: .3;
    }
  }
}