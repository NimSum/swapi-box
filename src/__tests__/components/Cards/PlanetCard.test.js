import { mount } from 'enzyme';
import React from 'react';
import { PlanetCard } from '../../../components/Cards/PlanetCard';

describe('Character card', () => {
  const mockupdateFavoriteCount = jest.fn();
  const mockCard = { 
    id: 123, 
    name: 'Earth', 
    terrain: 'Very Watery',
    climate: 'Hot or Cold', 
    population: '7500000000',
    imgNum: 1,
    residentNames: [ 
      { id: 1, name: 'Nimsum' },
      { id: 2, name: 'Dimsum' }
    ]
  };
  
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      < PlanetCard 
        card={ mockCard }
        updateFavoriteCount={ mockupdateFavoriteCount }/>
    )
  })

  it('Should match snapshot with card passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

})