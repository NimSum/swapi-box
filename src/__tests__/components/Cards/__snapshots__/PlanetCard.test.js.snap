// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character card Should match snapshot with card passed in 1`] = `
<PlanetCard
  card={
    Object {
      "climate": "Hot or Cold",
      "id": 123,
      "name": "Earth",
      "population": "7500000000",
      "residentNames": Array [
        Object {
          "id": 1,
          "name": "Nimsum",
        },
        Object {
          "id": 2,
          "name": "Dimsum",
        },
      ],
      "terrain": "Very Watery",
    }
  }
  updateFavoriteCount={[MockFunction]}
>
  <article
    className="planet-card"
    key="123"
  >
    <FavoriteCardBtn
      card={
        Object {
          "climate": "Hot or Cold",
          "id": 123,
          "name": "Earth",
          "population": "7500000000",
          "residentNames": Array [
            Object {
              "id": 1,
              "name": "Nimsum",
            },
            Object {
              "id": 2,
              "name": "Dimsum",
            },
          ],
          "terrain": "Very Watery",
        }
      }
      updateFavoriteCount={[MockFunction]}
    >
      <button
        className="favorite-card-btn"
        name="favorite-card"
        onClick={[Function]}
        title="Add to Favorites"
        type="button"
      >
        <img
          alt="card background"
          src="empire.png"
        />
        <img
          alt="card background"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "inactive-star.svg",
            }
          }
        />
        <img
          alt="card background"
          src="rebel-alliance.png"
        />
      </button>
    </FavoriteCardBtn>
    <h3>
      Earth
    </h3>
    <div
      className="terrain-container"
    >
      <h4>
        Terrain:
      </h4>
      <p>
        Very Watery
      </p>
    </div>
    <h4>
      Climate:
    </h4>
    <p>
      Hot or Cold
    </p>
    <h4>
      Population
    </h4>
    <p>
      7500000000
    </p>
    <div
      className="residents"
    >
      <h4>
        Residents:
      </h4>
      <p
        key="1"
      >
        Nimsum
      </p>
      <p
        key="2"
      >
        Dimsum
      </p>
    </div>
    <img
      alt="card background"
      className="bg-img"
      src="planet-5.png"
    />
  </article>
</PlanetCard>
`;
